src/components/Map/Map.tsx

```diff
@@ -71,49 +113,55 @@ export default function Map(props: MapProps): JSX.Element {
               };
             }),
           },
-        });
-
-        // add layer
-        map.current?.addLayer({
-          id: source.id,
-          type: 'fill',
-          source: source.name,
-          paint: {
-            'fill-color': source.fillColor,
-            'fill-opacity': source.fillOpacity,
+          layer: {
+            id: source.id,
+            type: 'fill',
+            paint: {
+              'fill-color': source.fillColor,
+              'fill-opacity': source.fillOpacity,
+            },
           },
-          filter: ['==', '$type', 'Polygon'],
-        });
-
-        // add click event handler
-        if (map.current !== null && source.onClick) {
-          const mapInstance = map.current;
-          const onClick = source.onClick;
-          mapInstance.on('click', source.id, (e) => {
-            new mapboxgl.Popup().setLngLat(e.lngLat).setHTML(onClick()).addTo(mapInstance);
-          });
-        }
-      });
-    });
-
-    // handle error
-    map.current.on('error', (event: any) => {
-      if (event?.error?.status === 401) {
-        // tslint:disable-next-line: no-console
-        console.error('Mapbox token expired');
-        if (onTokenExpired) {
-          onTokenExpired();
-        }
-      }
-    });
+        };
+      })
+      .filter((g) => g);
+    setGeoData(geo as any);
+    setLayerIds(geo.map((g: any) => g.layer.id));
+  }, [options, geoData, setGeoData, token]);
 
-    // fit to bounding box
-    map.current.fitBounds([bbox.lowerLeft, bbox.upperRight], { padding: 20 });
-  }, [token, onTokenExpired, options]);
+  useEffect(() => {
+    setMapKey(Date.now());
+  }, [token]);
 
   return (
-    <Box>
-      <Box ref={mapContainer} sx={{ height: '400px' }} />
+    <Box sx={{ display: 'flex', flexGrow: 1, height: '100%', minHeight: 100 }}>
+      <ReactMapGL
+        key={mapKey}
+        ref={mapRef}
+        mapboxAccessToken={token}
+        mapStyle='mapbox://styles/mapbox/satellite-v9'
+        interactiveLayerIds={layerIds}
+        onLoad={onMapLoad}
+        onError={onMapError}
+        onClick={onMapClick}
+      >
+        {geoData &&
+          (geoData as any[]).map((geo: any, index) => (
+            <Source type='geojson' key={index} data={geo.data}>
+              <Layer {...geo.layer} />
+            </Source>
+          ))}
+        <NavigationControl showCompass={false} style={navControlStyle} position='bottom-right' />
+        {popupInfo && (
+          <Popup
```

Comment:
I remember this from previous implementation, but do we need it here? I don't see popups on the design

Reply:
There is a popup in the withdrawal log. I kept popup code so it can be reused later for withdrawal log.
The data will be different. Will add an option now to enable popup.